test/Spec.hs

```diff
@@ -450,6 +451,17 @@ statefulTests =
                     ["default"]
          `shouldReturn` Just "12"
 
+doctypeTests :: SpecWith LarcenyHspecState
+doctypeTests = do
+  describe "doctypes" $ do
+    it "should render blank doctypes" $ do
+      "<doctype />" `shouldRenderM` "<!DOCTYPE html>"
```

Comment:
The thing I struggled with was, where do we insert the doctype? Larceny doesn't know anything about the total structure of the page, just whatever part it's rendering right now. I guess it could be outside of `process`, in `renderWith`:

```
renderWith :: Library s -> Substitutions s -> s -> Path -> IO (Maybe Text)
renderWith l sub s = (fmap .fmap) (flip (++) "<!DOCTYPE html>") (renderRelative l sub s [])
```

I haven't typechecked that but the general idea -- prepend the doctype after processing. Maybe simpler?